delfin/drivers/hitachi/hnas/nas_handler.py

```diff
@@ -0,0 +1,825 @@
+# Copyright 2021 The SODA Authors.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WarrayANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+import hashlib
+import time
+
+import six
+
+from oslo_log import log as logging
+
+from delfin import exception, utils
+from delfin.common import constants
+from delfin.drivers.utils.ssh_client import SSHPool
+from delfin.drivers.hitachi.hnas import constants as constant
+from delfin.drivers.utils.tools import Tools
+
+LOG = logging.getLogger(__name__)
+
+
+class NasHandler(object):
+
+    def __init__(self, **kwargs):
+        self.ssh_pool = SSHPool(**kwargs)
+        self.evs_list = []
+
+    @staticmethod
+    def get_size(limit, is_calculate=False):
+        if limit == '0B':
+            return 0
+        if limit == '-':
+            return 0 if is_calculate else '-'
+        return int(Tools.get_capacity_size(limit))
+
+    def ssh_do_exec(self, command_list):
+        res = ''
+        res = self.ssh_pool.do_exec_command(command_list)
+        while 'Failed to establish SSC connection' in res:
+            res = self.ssh_pool.do_exec_command(command_list)
+        return res
+
+    def login(self):
+        try:
+            result = self.ssh_do_exec(['cluster-show -y'])
+            if 'is not a recognized command' in result \
+                    or 'Unknown command' in result \
+                    or 'EVS' not in result:
+                raise exception.InvalidIpOrPort()
+        except Exception as e:
+            LOG.error("Failed to login netapp %s" %
+                      (six.text_type(e)))
+            raise e
+
+    @staticmethod
+    def split_value_map_list(
```

Comment:
What's this function for? Can not get the purpose of this function by the name and the implementation.

Reply:
Why use parameter `map_list` as return value instead of use `return` sentence

Reply:
has been  modify